mesa/datacollection.py

```diff
@@ -263,6 +338,9 @@ def collect(self, model):
         """Collect all the data for the given model object."""
         if self.model_reporters:
             for var, reporter in self.model_reporters.items():
+                # Add validation
+                self._validate_model_reporter(var, reporter, model)
+
```

Comment:
Does this imply that the validation is done every single time you try to collect the data? That seems inefficient and overkill.